packages/fiber/src/core/index.tsx

```diff
@@ -120,6 +120,27 @@ export type ReconcilerRoot<TCanvas extends Element> = {
   unmount: () => void
 }
 
+const logRecoverableError =
+  typeof reportError === 'function'
+    ? // In modern browsers, reportError will dispatch an error event,
+      // emulating an uncaught JavaScript error.
+      reportError
+    : (error: any) => {
+        // In older browsers and test environments, fallback to console.error.
+        console.error(error)
+      }
+
+type CreateContainerTypeFix = (
+  containerInfo: Parameters<typeof reconciler.createContainer>[0],
+  tag: Parameters<typeof reconciler.createContainer>[1],
+  hydrationCallbacks: Parameters<typeof reconciler.createContainer>[3],
+  isStrictMode: boolean,
+  concurrentUpdatesByDefaultOverride: any,
+  identifierPrefi: any,
+  onRecoverableError: any,
+  transitionCallbacks: any,
+) => ReturnType<typeof reconciler.createContainer>
```

Comment:
Add a comment here explaining why we need this and when we can remove ( with maybe a link to an issue to remove it )

```js
/**
 * 
 * In some cases (that are still not really clear to me), when an error occur when rendering react-reconciler might fire onRecoverableError, which is a new parameter in createContainer. Unfortunately the typings doesn't yet support the new parameters, so they have to be monkey patched.
 * @see https://github.com/pmndrs/react-three-fiber/pull/2261
 * @see https://github.com/facebook/react/pull/23207
 * @todo *linkToTheIssue*
 */
```

Reply:
@[giulioz](https://github.com/giulioz) mentioned on discord that this is part of what the reconciler should naturally have. react tries to call onRecoverableError and if it is undefined it crashes. the reason why onRecoverableError needs to be called in the first place can be any.

Reply:
Still, add the comment before merging, otherwise it's another "thing" that few understand

Reply:
Got it in https://github.com/pmndrs/react-three-fiber/commit/32c87aebacca3bce0229422fd125ef1713a12206.